mesonbuild/scripts/depfixer.py

```diff
@@ -337,20 +338,61 @@ def remove_rpath_entry(self, entrynum):
             entry.write(self.bf)
         return None
 
+def fix_elf(fname, new_rpath, verbose=True):
+    with Elf(fname, verbose) as e:
+        if new_rpath is None:
+            e.print_rpath()
+            e.print_runpath()
+        else:
+            e.fix_rpath(new_rpath)
+
+def get_darwin_rpaths_to_remove(fname):
+    out = subprocess.check_output(['otool', '-l', fname], universal_newlines=True)
+    result = []
+    current_cmd = 'FOOBAR'
+    for line in out.split('\n'):
+        if ' ' not in line:
+            continue
+        key, value = line.strip().split(' ', 1)
+        if key == 'cmd':
+            current_cmd = value
+        if key == 'path' and current_cmd == 'LC_RPATH':
+            rp = value.split('(', 1)[0].strip()
+            result.append(rp)
+    return result
+
+def fix_darwin(fname, new_rpath):
+    try:
+        for rp in get_darwin_rpaths_to_remove(fname):
+            subprocess.check_call(['install_name_tool', '-delete_rpath', rp, fname])
+        if new_rpath != '':
+            subprocess.check_call(['install_name_tool', '-add_rpath', new_rpath, fname])
+    except Exception as e:
+        raise
+        sys.exit(0)
+
+def fix_rpath(fname, new_rpath, verbose=True):
+    try:
+        fix_elf(fname, new_rpath, verbose)
+        return 0
+    except SystemExit as e:
+        if isinstance(e.code, int) and e.code == 0:
+            pass
+        else:
+            raise
+    if shutil.which('install_name_tool'):
+        fix_darwin(fname, new_rpath)
+    return 0
+
 def run(args):
     if len(args) < 1 or len(args) > 2:
         print('This application resets target rpath.')
         print('Don\'t run this unless you know what you are doing.')
         print('%s: <binary file> <prefix>' % sys.argv[0])
         sys.exit(1)
-    with Elf(args[0]) as e:
-        if len(args) == 1:
-            e.print_rpath()
-            e.print_runpath()
-        else:
-            new_rpath = args[1]
-            e.fix_rpath(new_rpath)
-    return 0
+    fname = args[0]
```

Comment:
__[Flake8]__

[F841] local variable 'fname' is assigned to but never used


<!-- Comment by [SideCI](https://sideci.com) -->